synapse/models/infotech.py

```diff
@@ -77,3 +58,170 @@ def getBaseModels():
         }
         name = 'it'
         return ((name, modl), )
+
+    @on('node:form', form='it:dev:str')
+    def _onFormItDevStr(self, mesg):
+        props = mesg[1].get('props')
+        props['it:dev:str:norm'] = mesg[1].get('valu').lower()
+
+    @modelrev('it', 201708161120)
+    def _revModl201708161120(self):
+        self.core.loadDataModel({
+            'types':(
+
+                ('it:dev:str', {'subof':'str', 'doc':'A developer selected string'}),
+                ('it:dev:int', {'subof':'int', 'doc':'A developer selected int constant'}),
```

Comment:
Should these be comp types of (name, val)?

Ex:
```
const int COOLINT = 42;
string COOLSTR{ u8"c++14" };
```

Do we care about variable names from source code? or just the values?